lib/cucumber/glue/registry_and_more.rb

```diff
@@ -207,7 +207,17 @@ def parameter_type_envelope(parameter_type)
             name: parameter_type.name,
             regular_expressions: parameter_type.regexps.map(&:to_s),
             prefer_for_regular_expression_match: parameter_type.prefer_for_regexp_match?,
-            use_for_snippets: parameter_type.use_for_snippets?
+            use_for_snippets: parameter_type.use_for_snippets?,
+            source_reference: Cucumber::Messages::SourceReference.from_h(
+              # TODO: This needs fixing before we release the next version - this is simply put in to trick the
```

Comment:
@jkronegg - If you want to mob on this or provide extra context / solutions before I'm next back feel free.

Reply:
@luke-hill thanks for asking, but I'm not a ruby developer. What do you expect from me?

Reply:
Ah sorry, nevermind then. I'll look into this later on. I thought you might have known what to do as you fixed the other ones I believe.

Reply:
ping @botandrose if you want something technical to get your teeth into, this is something which I have half-fixed, but incorrectly. I haven't had much chance to fix this and I'm busy fixing a few other things in CCK / rails, so feel free to tackle if you want.

Reply:
@luke-hill Wow, this is some meta meta stuff!

I took a peek and managed to get something working, but it ain't pretty:

```ruby
uri, line = *parameter_type.instance_variable_get(:@transformer).source_location
...
source_reference: Cucumber::Messages::SourceReference.new(
  uri: uri,
  location: Cucumber::Messages::Location.new(
    line: line
  )
)
```

If this is the right direction, we should probably push that first line into `Cucumber::Expressions::ParameterType#source_location`, or at least expose a `#transformer` reader.

Reply:
Atm I'm doing some refactoring for a couple of days. I have some more commits on this branch. So if you'd like keep digging on your own, and then I'll push up an update here thurs/fri.

For clarity, this is to enable us to be more conformant to the new CCK (Which we're tracking heavily behind). If you check the person above they made changes in Java for this same thing here. See https://github.com/cucumber/cucumber-ruby/issues/1702 initially for the original issue and https://github.com/cucumber/cucumber-jvm/pull/2739/files for the code solution.

I'm not expecting this to be a big fix. But if you're interested to keep digging feel free. I'll be able to assist more next week.

Reply:
@botandrose I've fixed this.